go-mode.el

```diff
@@ -1670,7 +1670,7 @@ with goflymake \(see URL `https://github.com/dougm/goflymake'), gocode
 
   (setq imenu-generic-expression
         '(("type" "^type *\\([^ \t\n\r\f]*\\)" 1)
-          ("func" "^func *\\(.*\\) {" 1)))
+          ("func" "^func *\\([^{]*\\) {" 1)))
```

Comment:
Why do we have to match beyond the function name? Does imenu do something with the args or return values?

Reply:
I menu just displays what you match. I think the original regex was written that way to properly match both package level functions and methods on structs. I tried to maintain what we were already matching while supporting a split across newlines.

Reply:
If that is the case then your new regex breaks on signatures like:
```go
func foo(s struct { i int }, i interface { bar() }) {
```

But, based on my testing, it isn't the case. It seems like the regex only needs to match the function name.